game-app/game-core/src/main/java/games/strategy/triplea/ResourceLoader.java

```diff
@@ -173,33 +168,21 @@ public boolean hasPath(final String path) {
     return loader.resources(searchPath).findFirst();
   }
 
-  /**
-   * Returns an input stream for the specified resource or {@code null} if the resource does not
-   * exist. The caller is responsible for closing the returned input stream.
-   *
-   * @throws IllegalStateException If the specified resource exists but the input stream cannot be
-   *     opened.
-   */
-  public @Nullable InputStream getResourceAsStream(final String path) {
-    final URL url = getResource(path);
-    if (url == null) {
-      return null;
-    }
-
-    return UrlStreams.openStream(url)
-        .orElseThrow(() -> new IllegalStateException("Failed to open an input stream to: " + path));
-  }
-
-  public ThrowingSupplier<InputStream, IOException> optionalResource(final String path) {
-    return () ->
-        Optional.ofNullable(getResourceAsStream(path))
-            .orElseGet(() -> new ByteArrayInputStream(new byte[0]));
+  public Optional<Path> optionalResource(final String path) {
+    return findResource(path)
```

Comment:
A future change could have `findResource` return a `Path`? Yeah, the URL is no longer needed, it was used to be able to index into zip files, we no longer need to do that.

Reply:
Potentially. I only had a brief look at usages outside of my changes where URLs are used in a lot of places most of them can probably get migrated.
However the resource-loading itself uses UrlClassLoader to load files. It's really convenient because you can pass multiple directories/jar/zip-files as a source which automatically takes care of falling back to our assets we ship with triplea but preferring any assets that come with the map directly. This API only returns URLs though so we're stuck with that for a while unless we want to rewrite the resource loading code.
That being sad I'm sure there's a way to create a "FileSystem" using the nio APIs which has the same effect and creates a safe view of resources in a virtual filesystem

Reply:
Ok. To not get have the point be lost, I think most of the URL's we can replace with simple Path objects. I don't think that should require too much rewriting, mostly replacing of object types. The URLs do get converted to file system paths eventually, so should be tractable.